### PR TITLE
test(p/avl): increase test coverage

### PR DESCRIPTION
**Pull request by notJoon**
# Description

I added tests to the AVL package.
Checking in Go, the coverage comes out to be 88.6%.

스크린샷 2024-04-03 오후 3 57 14

### CODE REVIEW

File: examples/gno.land/p/demo/avl/node.gno [hunk 1]
```diff
@@ -167,51 +174,52 @@ func (node *Node) Remove(key string) (
...
+		return node, "", nil, false
+	}
+	if key < node.key {
+		var newLeftNode *Node
+		newLeftNode, newKey, value, removed = node.getLeftNode().Remove(key)
+		if !removed {
+			return node, "", value, false
+		} else if newLeftNode == nil { // left node held value, was removed
+			return node.rightNode, node.key, value, true
 		}
```
Comment (harry-hov) [inline examples/gno.land/p/demo/avl/node.gno:187]:
```suggestion
		if !removed {
			return node, "", value, false
		}
		if newLeftNode == nil { // left node held value, was removed
			return node.rightNode, node.key, value, true
		}
```

**Review by harry-hov — APPROVED** [review]
Left few suggestions. Rest LGTM 💯 

Thank you for doing this. 🙏

File: examples/gno.land/p/demo/avl/node.gno [hunk 2]
```diff
@@ -65,16 +73,20 @@ func (node *Node) Has(key string) (has bool) {
 	} else {
 		if key < node.key {
 			return node.getLeftNode().Has(key)
-		} else {
-			return node.getRightNode().Has(key)
 		}
+		return node.getRightNode().Has(key)
 	}
```
[the remark this answers is not in the document]
Comment (notJoon) [reply examples/gno.land/p/demo/avl/node.gno:78]: I thought I had removed `else` as much as possible, but I didn't see this 😅

**Comment by notJoon** [comment]
> Left few suggestions.

Thank you for the review! I have resolved all the parts you commented on.

Heres the additional modifications:
- extracted the leaf node insertion logic from the `Set` function and created a separate `setLeaf` function.
- remove `stop` variable from the `TraverseInRange` function.

File: examples/gno.land/p/demo/avl/node.gno [hunk 3]
```diff
@@ -321,43 +344,36 @@ func (node *Node) TraverseInRange(start, end string, ascending bool, leavesOnly
...
 
 	if ascending {
 		// check lower nodes, then higher
-		if afterStart {
-			stop = node.getLeftNode().TraverseInRange(start, end, ascending, leavesOnly, cb)
-		}
-		if stop {
-			return stop
+		if afterStart && node.getLeftNode().TraverseInRange(start, end, ascending, leavesOnly, cb) {
```
Comment (harry-hov) [inline examples/gno.land/p/demo/avl/node.gno:360]: can we keep the `stop` for sake of readability?

right now it take too much of my brain cycles to understand the conditions.

**Comment by harry-hov** [comment]
Added one last comment from my side. Otherwise we are good to merge 🙏

File: examples/gno.land/p/demo/avl/node.gno (hunk 3 above)
Comment (notJoon) [reply examples/gno.land/p/demo/avl/node.gno]: reverted https://github.com/gnolang/gno/pull/1881/commits/1a6845c25ac77459ba7588b676016dbe79bbff70

**Pull request merged**